samples/matter/common/src/binding_handler.cpp

```diff
@@ -5,83 +5,92 @@
  */
 
 #include "binding_handler.h"
-#include "temp_sensor_manager.h"
-#include "temperature_measurement/sensor.h"
-
-#include <app-common/zap-generated/ids/Clusters.h>
-#include <app-common/zap-generated/ids/Commands.h>
-#include <app/CommandSender.h>
-#include <app/OperationalSessionSetup.h>
-#include <app/clusters/bindings/BindingManager.h>
-#include <app/clusters/bindings/bindings.h>
-#include <app/data-model/Decode.h>
-#include <app/util/binding-table.h>
-#include <controller/InvokeInteraction.h>
-#include <controller/ReadInteraction.h>
-#include <lib/support/CodeUtils.h>
+
+#include <zephyr/logging/log.h>
 
 LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
 
 using namespace chip;
 using namespace chip::app;
 
-void BindingHandler::Init()
+CHIP_ERROR BindingHandler::Init(deviceChangedHandler pHandler)
 {
-	DeviceLayer::PlatformMgr().ScheduleWork(InitInternal);
+	CHIP_ERROR error      = CHIP_NO_ERROR;
```

Comment:
```suggestion
	CHIP_ERROR error;
```
or
```suggestion
	CHIP_ERROR error{};
```

Reply:
I guess we don't need it at all. Let's just return prematurely with direct `CHIP_ERROR_INVALID_ARGUMENT` if null and `CHIP_NO_ERROR` in case of success. You can use `VerifyOrRetrurnError` macro for that.